ihp-ide/IHP/SchemaCompiler.hs

```diff
@@ -336,20 +336,44 @@ primaryKeyTypeName :: Text -> Text
 primaryKeyTypeName name = "Id' " <> tshow name <> ""
 
 compileData :: (?schema :: Schema) => CreateTable -> Text
-compileData table@(CreateTable { name, columns }) =
-        "data " <> modelName <> "' " <> typeArguments
-        <> " = " <> modelName <> " {"
+compileData table@(CreateTable { name, inherits }) =
+    "data " <> modelName <> "' " <> typeArguments <> " = " <> modelName <> " {"
+    <>
+        parentFields
         <>
-            table
-            |> dataFields
-            |> map (\(fieldName, fieldType) -> fieldName <> " :: " <> fieldType)
-            |> commaSep
-        <> "} deriving (Eq, Show)\n"
+        table
+        |> dataFields
+        |> map (\(fieldName, fieldType) -> fieldName <> " :: " <> fieldType)
+        |> commaSep
+    <> "} deriving (Eq, Show)\n"
     where
         modelName = tableNameToModelName name
         typeArguments :: Text
         typeArguments = dataTypeArguments table |> unwords
 
+        -- If the table inherits from another table, include the fields from the parent table.
+        parentFields = inherits
+            |> maybe "" (\parentTable -> compileParentFields parentTable)
+
+            -- Add comma, if there are fields from parent tables
+            |> (\parentFields -> if null parentFields then "" else parentFields <> ", ")
+
+        compileParentFields parentTable =
+            let parentTableDef = findTableByName parentTable
+            in parentTableDef
+                |> maybe [] (dataFields . (.unsafeGetCreateTable))
+                -- Remove the MetaBag field from the parent table.
+                -- @todo: Avoid clashing of field names.
+                |> filter (\(fieldName, _) -> fieldName /= "meta")
+                |> map (\(fieldName, fieldType) -> fieldName <> " :: " <> fieldType)
+                |> commaSep
```

Comment:
@mpscholten With this code I'm getting inside `Generated/ActualTypes.hs`

```
data PostRevision' postId = PostRevision {id :: (Id' "posts"), title :: Text, body :: Text, userId :: userId, postRevisions :: postRevisions, revisionId :: (Id' "post_revisions"), postId :: postId, meta :: MetaBag} deriving (Eq, Show)
```

`postRevisions :: postRevisions` is wrong in this case. Not sure how to know when to filter it out

Reply:
I think you need to add a type argument for `postRevisions`:

```
-- current:
data PostRevision' postId = 

-- fixed:
data PostRevision' postId postRevisions = 
```